api_proxy/Dockerfile

```diff
@@ -0,0 +1,8 @@
+# How to build:
+# docker run --rm -e BUILD_GOOS="linux" -e BUILD_GOARCH="amd64" -v $(pwd):/src \
+#   -v /var/run/docker.sock:/var/run/docker.sock \
+#   centurylink/golang-builder-cross gcr.io/$DOCKER_NAMESPACE/api-proxy
```

Comment:
I'd like to be able to use vendored deps in the future, but it isn't required now, which should be hermetic.
I reworded the build instructions to include a better mix of "how to build".